src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs

```diff
@@ -564,18 +599,41 @@ private async Task<BitDataGridItemsProviderResult<Country>> Get7CountriesAsync()
     BitDataGridPaginationState pagination = new() { ItemsPerPage = 15 };
     IQueryable<Country> items;
     IQueryable<Country> data;
-    string nameFilter1 = string.Empty;
-    string nameFilter2 = string.Empty;
+    IQueryable<FoodRecall> foodRecallData;
+    string typicalSampleNameFilter = string.Empty;
+    string virtualSampleNameFilter = string.Empty;
     BitDataGridSort<Country> rankSort = BitDataGridSort<Country>.ByDescending(x => x.Medals.Gold).ThenDescending(x => x.Medals.Silver).ThenDescending(x => x.Medals.Bronze);
 
-    IQueryable<Country> FilteredItems => items?.Where(x => x.Name.Contains(nameFilter1, StringComparison.CurrentCultureIgnoreCase));
-    IQueryable<Country> VirtualFilteredItems => items?.Where(x => x.Name.Contains(nameFilter2, StringComparison.CurrentCultureIgnoreCase));
-
+    IQueryable<Country> FilteredItems => items?.Where(x => x.Name.Contains(typicalSampleNameFilter, StringComparison.CurrentCultureIgnoreCase));
+    
     protected override async Task OnInitializedAsync()
     {
         items = (await GetCountriesAsync(0, null, null, true, CancellationToken.None)).Items.AsQueryable();
         data = (await Get7CountriesAsync()).Items.AsQueryable();
+        foodRecallData= (await GetfoodRecallAsync(0, null, null, true,virtualSampleNameFilter, CancellationToken.None)).Items.AsQueryable();
+    }
 
+    private async Task<BitDataGridItemsProviderResult<FoodRecall>> GetfoodRecallAsync(int startIndex, int? count, string sortBy, bool sortAscending,string search, CancellationToken cancellationToken)
+    {
+        var url = NavManager.GetUriWithQueryParameters("http://api.fda.gov/food/enforcement.json", new Dictionary<string, object?>
+            {
+                { "skip", startIndex },
+                { "limit", count },
+                { "search", search },
+            });
+
+        using var httpResponse = await Http.GetAsync(url, cancellationToken);       
+
+        if (!httpResponse.IsSuccessStatusCode)
+        {
+            return new BitDataGridItemsProviderResult<FoodRecall>() { Items=new List<FoodRecall>() { } , TotalItemCount=0 };
```

Comment:
Use System.Text.Json Source Generator to improve the performance of this (and the rest of http calls if any)